subprojects/workers/src/main/java/org/gradle/workers/internal/ClassLoaderStructureProvider.java

```diff
@@ -16,61 +16,36 @@
 
 package org.gradle.workers.internal;
 
-import com.google.common.cache.Cache;
-import com.google.common.cache.CacheBuilder;
 import org.gradle.initialization.ClassLoaderRegistry;
 import org.gradle.initialization.MixInLegacyTypesClassLoader;
-import org.gradle.internal.UncheckedException;
 import org.gradle.internal.classloader.FilteringClassLoader;
 import org.gradle.internal.classloader.VisitableURLClassLoader;
-import org.gradle.internal.classpath.DefaultClassPath;
 
 import java.io.File;
-import java.util.concurrent.Callable;
-import java.util.concurrent.ExecutionException;
 
 public class ClassLoaderStructureProvider {
     private final ClassLoaderRegistry classLoaderRegistry;
-    private final Cache<Iterable<File>, ClassLoaderStructure> knownClassLoaderStructures;
```

Comment:
This was a hotspot at one point, but it was likely because of things we've since removed (like calculating the api spec every time instead of reusing it from the registry).  We should keep an eye on perf tests and make sure they don't wobble any after removing this.